k2/csrc/fsa_utils.cu

```diff
@@ -168,7 +204,21 @@ static Fsa TransducerFromStream(std::string first_line, std::istringstream &is,
       float score = scale * StringToFloat(splits[4]);
       arcs.emplace_back(src_state, dest_state, symbol, score);
       state_aux_labels.push_back(aux_label);
+      max_state = std::max(max_state, std::max(src_state, dest_state));
+    } else if (num_fields == 2u) {
+      //   0            1
+      // final_state  score
+      // In this case, openfst is true. There could be multiple final states, so
+      // we first have to collect all the final states, and then work out the
+      // super final state.
+      K2_CHECK(openfst) << "Invalid line: " << line
+                        << "\nFinal state with weight detected in K2 format";
+      original_final_states.push_back(StringToInt(splits[0]));
+      original_final_weights.push_back(StringToFloat(splits[1]));
+      max_state = std::max(max_state, original_final_states.back());
```

Comment:
With `std::max`, I think you assume that the first column is not in non-decreasing order,
otherwise, you would use `original_final_states.back()` as the max state.